### PR TITLE
Support code coverage

### PR DESCRIPTION
**Pull request by yannbf**
Issue: #116

## Setting up code coverage

The test runner supports code coverage with the `--coverage` flag or `STORYBOOK_COLLECT_COVERAGE` environment variable. The pre-requisite is that your components are instrumented using [istanbul](https://istanbul.js.org/).

### Instrument the code

Given that your components' code runs in the context of a real browser, they have to be instrumented so that the test runner is able to collect coverage. In order to do so, you have to setup the instrumentation yourself.

Install the istanbul babel plugin:

```sh
yarn add -D babel-plugin-istanbul
```

Storybook allows code transpilation with babel out of the box by configuring the `babel` function in your `main.js`. Add the `istanbul` plugin:

```js
// .storybook/main.js
module.exports = {
  // ...rest of your code here
  babel: async (options) => {
    options.plugins.push([
      'istanbul',
      {
        // provide include patterns if you like
        include: ['src/components/**'],
        // provide exclude patterns if you like
        exclude: [
          '**/*.d.ts',
          '**/*{.,-}{spec,stories,types}.{js,jsx,ts,tsx}',
        ],
      },
    ]);

    return options;
  },
};
```

The babel plugin has default options that might suffice to your project, however if you want to know which options are taken into account you can check them [here](https://github.com/istanbuljs/babel-plugin-istanbul/blob/master/src/index.js#L82).

### Run tests with --coverage flag

After setting up instrumentation, run Storybook then run the test-runner with `--coverage`:

```sh
yarn test-storybook --coverage
```

The test runner will report the results in the CLI and generate a `.nyc_output/coverage.json` file which can be used by `nyc`.

<details>
  <summary>📦 Published PR as canary version: <code>0.2.0--canary.120.86b6b43.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/test-runner@0.2.0--canary.120.86b6b43.0
  # or 
  yarn add @storybook/test-runner@0.2.0--canary.120.86b6b43.0
  ```
</details>

### CODE REVIEW

File: bin/test-storybook.js [hunk 1]
```diff
@@ -162,6 +171,10 @@ const main = async () => {
 
   process.env.TARGET_URL = targetURL;
 
+  if (runnerOptions.coverage) {
+    process.env.COLLECT_COVERAGE = 'true';
```
Comment (shilman) [inline bin/test-storybook.js:175]: I'm thinking maybe we should prefix these variables, e.g. `SB_COLLECT_COVERAGE`. WDYT?

**Review by shilman — APPROVED** [review]
LGTM!

**Review by shilman — CHANGES_REQUESTED** [review]
Also shouldn't this be a minor release?

File: bin/test-storybook.js (hunk 1 above)
Comment (yannbf) [reply bin/test-storybook.js]: makes sense as discussed. Done!

**Review by shilman — APPROVED** [review]

**Pull request merged**